test/hotspot/jtreg/compiler/c2/cr7200264/TestIntVect.java

```diff
@@ -21,622 +21,763 @@
+    void test_sllc_n(int[] a0, int[] a1) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]<<(-VALUE));
+        }
+    }
+
+    void test_sllc_o(int[] a0, int[] a1) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]<<SHIFT);
+        }
+    }
+
+    void test_sllc_on(int[] a0, int[] a1) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]<<(-SHIFT));
+        }
+    }
+
+    @Test
+    @IR(counts = { IRNode.LSHIFT_VI, ">= 1" },
+        applyIfCPUFeature = {"sse2", "true"})
+    void test_sllv(int[] a0, int[] a1, int b) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]<<b);
+        }
+    }
+
+    @Test
+    @IR(counts = { IRNode.URSHIFT_VI, ">= 1" },
+        applyIfCPUFeature = {"sse2", "true"})
+    void test_srlc(int[] a0, int[] a1) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]>>>VALUE);
+        }
+    }
+
+    @Test
+    @IR(counts = { IRNode.URSHIFT_VI, ">= 1" },
+        applyIfCPUFeature = {"sse2", "true"})
+    void test_srlc_n(int[] a0, int[] a1) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]>>>(-VALUE));
+        }
+    }
+
+    void test_srlc_o(int[] a0, int[] a1) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]>>>SHIFT);
+        }
+    }
+
+    void test_srlc_on(int[] a0, int[] a1) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]>>>(-SHIFT));
+        }
+    }
+
+    @Test
+    @IR(counts = { IRNode.URSHIFT_VI, ">= 1" },
+        applyIfCPUFeature = {"sse2", "true"})
+    void test_srlv(int[] a0, int[] a1, int b) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]>>>b);
+        }
+    }
+
+    @Test
+    @IR(counts = { IRNode.RSHIFT_VI, ">= 1" },
+        applyIfCPUFeature = {"sse2", "true"})
+    void test_srac(int[] a0, int[] a1) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]>>VALUE);
+        }
+    }
+
+    @Test
+    @IR(counts = { IRNode.RSHIFT_VI, ">= 1" },
+        applyIfCPUFeature = {"sse2", "true"})
+    void test_srac_n(int[] a0, int[] a1) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]>>(-VALUE));
+        }
+    }
+
+    void test_srac_o(int[] a0, int[] a1) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]>>SHIFT);
+        }
+    }
+
+    void test_srac_on(int[] a0, int[] a1) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]>>(-SHIFT));
+        }
+    }
+
+    @Test
+    @IR(counts = { IRNode.RSHIFT_VI, ">= 1" },
+        applyIfCPUFeature = {"sse2", "true"})
+    void test_srav(int[] a0, int[] a1, int b) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]>>b);
+        }
+    }
+
+    void test_pack2(long[] p2, int[] a1) {
```

Comment:
Intersting pattern! But that would require optimizations we currently do not have.
Maybe we can do that in the future, with heavy upgrades to the autovectorizer,
or other optimizations that merge loads / stores.

Reply:
applies to all pack/swap cases here